### PR TITLE
Added softmax activation function

### PR DESCRIPTION
**Pull request by huntrar**
Added softmax activation function + derivative

### CODE REVIEW

**Comment by jorgenkg** [comment]
Hello @huntrar,
The derivate does not seem correct. Have you tested your commit?

**Comment by huntrar** [comment]
My mistake, you are definitely correct. After checking it seems the derivate should be equivalent to that of the sigmoid function. I have reflected this change now.

**Comment by jorgenkg** [comment]
The new derivative is only valid if combined with a cross-entropy cost function implemented with a "special case" derivative. Please refer to the image below:

her

While the general form and derivative of the cross entropy function takes the form:

der


Image credits: [CSC2515 (Fall 2007) lectured by Geoff Hinton and Rich Zemel](https://www.cs.nyu.edu/~roweis/csc2515/)

**Comment by huntrar** [comment]
Apologies again, it should be clear at this point I am new to this. I will work on correcting this derivative.

In the meantime, I have added MSE to cost_functions. Does it seem correct to you?

**Comment by huntrar** [comment]
It seems incorrect actually, the error remained constant while testing.

Closing this PR, may resubmit another when I am more confident in my changes...
